torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -205,206 +268,90 @@ void replaceWithIValue(Value* v, IValue val) {
-          // from y to z. To do this we try to associate symbolic dimensions
-          // or concrete sizes with the integer list inputs that have a
-          // constructor taken from constants or y.size() or y.size(0)
-          auto list_construct = node_->input(node_index)->node();
-          std::vector<ShapeArg> shape;
-          for (Value* v : list_construct->inputs()) {
-            if (auto constant = constant_as<int64_t>(v)) {
-              shape.emplace_back(*constant);
-            } else if (v->node()->kind() == aten::size) {
-              auto const_index = constant_as<int64_t>(v->node()->input(1));
-              auto tt = v->node()->input(0)->type()->expect<TensorType>();
-              auto ss = tt->symbolic_sizes();
-              if (!ss.rank() || !const_index) {
-                // if we are getting a size of a tensor, it is an unknown
-                // symbolic dimension instead of an unknown integer (must be
-                // >=0)
-                shape.emplace_back(at::ShapeSymbol::newSymbol());
-                continue;
-              }
-              auto norm_index = normIndex(*const_index, *ss.rank());
-              if (!norm_index) {
-                shape.emplace_back(at::ShapeSymbol::newSymbol());
-                continue;
-              }
-              shape.emplace_back(ss[*norm_index]);
-            } else {
-              shape.emplace_back(ShapeArg::unknownInteger());
-            }
-          }
-          node_symbolic_input_indices_.emplace_back(
-              graph_index, std::move(shape));
-        } else if (
-            node_->input(node_index)->node()->kind() == aten::size &&
-            !db.hasWriters(node_->input(node_index))) {
-          auto ten_inp = node_->input(node_index)->node()->input();
-          auto ss = ten_inp->type()->expect<TensorType>()->symbolic_sizes();
-          node_symbolic_input_indices_.emplace_back(graph_index, ss);
-        }
-      }
-    }
+std::ostream& operator<<(std::ostream& out, const SsaArgument& sa) {
+  if (const IValue* iv = c10::get_if<IValue>(&sa)) {
+    out << *iv;
+  } else {
+    out << c10::get<ShapeArguments>(sa);
   }
+  return out;
+}
 
-  void addTensorInputMetaData(
-      Value* tensor_v,
-      size_t shape_compute_graph_index) {
-    auto tt = tensor_v->type()->expect<TensorType>();
-    // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
-    c10::SymbolicShape symbolic_shapes = tt->symbolic_sizes();
+struct SymbolicShapeOpAnalyzer {
+  std::shared_ptr<Graph> shape_compute_graph_;
+  const FunctionSchema* schema_;
+  std::vector<SsaArgument> inputs_;
 
-    // for testing, we don't insert complete tensor shapes and rely on our
-    // partial evaluation pipeline to propagate information.
-    // this is a good proxy for our ability to propagate non-complete shape
-    // information.
+  void substituteInput(SsaArgument& argument, Value* graph_in_var) {
```

Comment:
How does this work in the path where there is no `Value*` ?

Reply:
Do you mean that there can be arguments into a JIT graph that are not a Value* ?